livre/core/render/SelectVisibles.h

```diff
@@ -0,0 +1,167 @@
+/* Copyright (c) 2015, EPFL/Blue Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef LIVRE_SELECTVISIBLES_H
+#define LIVRE_SELECTVISIBLES_H
+
+#include <livre/lib/types.h>
+#include <livre/core/data/LODNode.h>
+#include <livre/core/maths/maths.h>
+#include <livre/core/data/VolumeInformation.h>
+#include <livre/core/visitor/DataSourceVisitor.h>
+#include <livre/core/visitor/VisitState.h>
+#include <livre/core/render/Frustum.h>
+
+//#define LIVRE_STATIC_DECOMPOSITION
+
+namespace livre
+{
+/** Selects all visible rendering nodes */
+class SelectVisibles : public DataSourceVisitor
+{
+public:
+    SelectVisibles( const DataSource& dataSource,
+                    const Frustum& frustum,
+                    const uint32_t windowHeight,
+                    const float screenSpaceError,
+                    const uint32_t minLOD,
+                    const uint32_t maxLOD,
+                    const Range& range )
+    : DataSourceVisitor( dataSource )
+    , _frustum( frustum )
+    , _windowHeight( windowHeight )
+    , _screenSpaceError( screenSpaceError )
+    , _minLOD( minLOD )
+    , _maxLOD( maxLOD )
+    , _range( range )
+    {}
+
+    const NodeIds& getVisibles() const { return _visibles; }
+
+protected:
+    void visitPre() final { _visibles.clear(); }
+
+    void visit( const LODNode& lodNode, VisitState& state ) final
+    {
+        const Boxf& worldBox = lodNode.getWorldBox();
+        const bool isInFrustum = _frustum.boxInFrustum( worldBox );
+        if( !isInFrustum )
```

Comment:
I don't know if it's better... But you can collapse these two line:
if( !_frustum.boxInFrustum( worldBox ))